airflow/models/dag.py

```diff
@@ -3745,6 +3758,7 @@ def dag(
     owner_links: dict[str, str] | None = None,
     auto_register: bool = True,
     fail_stop: bool = False,
+    default_run_id: Callable | None = None,
```

Comment:
+1 here - Can you add some type hints so that mypy can check the signature if the callable is customized?
```suggestion
    default_run_id: Callable[[DagRunType, DateTime, DataInterval, dict], str] | None = None,
```